lib/autoBlock.js

```diff
@@ -0,0 +1,215 @@
+var _ = require('lodash')
+var async = require('async')
+var httpStatus = require('http-status')
+
+var utils = require('./utilities')
+
+// TODO reject invalid options/flags (e.g., errorDeafults typos)
+
+// Runs a block using common block/controller pattern
+function runBlock (config, done) {
+  // Set config.done if not provided
+  if (!config.done) {
+    config.done = done || (() => {
+    })
```

Comment:
I think the if checked is redundant if you use the || like below. Also, I think the lambda could be a single line:
```
config.done = done || () => { }
```